# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../linux24-psionw/pro16.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -ur linux24-psionw/arch/arm/config.in linux24-psionw-16pro/arch/arm/config.in
--- linux24-psionw/arch/arm/config.in	2000-01-01 00:00:00.000000000 +0100
+++ linux24-psionw-16pro/arch/arm/config.in	2006-07-29 22:47:18.000000000 +0200
@@ -188,6 +188,7 @@
 comment 'Psion Windermere Implementations'
 choice 'Psion-Windermere Implementations'                    \
         "Psion-5MX-16MB-and-Ericsson-MC218 CONFIG_PSIONW_5MX \
+         Psion-5MX-Pro-16MB CONFIG_PSIONW_5MXPRO16MB         \
          Psion-5MX-Pro-24MB CONFIG_PSIONW_5MXPRO24MB         \
          Psion-5MX-Pro-32MB CONFIG_PSIONW_5MXPRO32MB         \
          Psion-Revo-8MB CONFIG_PSIONW_REVO                   \
diff -ur linux24-psionw/arch/arm/mach-psionw/psionw-arch.c linux24-psionw-16pro/arch/arm/mach-psionw/psionw-arch.c
--- linux24-psionw/arch/arm/mach-psionw/psionw-arch.c	2000-01-01 00:00:00.000000000 +0100
+++ linux24-psionw-16pro/arch/arm/mach-psionw/psionw-arch.c	2006-07-29 22:49:02.000000000 +0200
@@ -48,7 +48,7 @@
 	     char **cmdline, struct meminfo *mi)
 {
 
-#ifdef CONFIG_PSIONW_5MX
+#if defined(CONFIG_PSIONW_5MX) || defined(CONFIG_PSIONW_5MXPRO16MB)
 	SET_BANK(0, 0xc0000000, 8 * 1024 * 1024);
 	SET_BANK(1, 0xc1000000, 8 * 1024 * 1024);
 	mi->nr_banks = 2;
diff -ur linux24-psionw/drivers/pcmcia/psion_etna.h linux24-psionw-16pro/drivers/pcmcia/psion_etna.h
--- linux24-psionw/drivers/pcmcia/psion_etna.h	2000-01-01 00:00:00.000000000 +0100
+++ linux24-psionw-16pro/drivers/pcmcia/psion_etna.h	2006-07-29 22:46:38.000000000 +0200
@@ -72,7 +72,7 @@
 
 	save_flags_cli(flags);
 
-#if defined(CONFIG_PSIONW_5MXPRO24MB) | defined(CONFIG_PSIONW_5MXPRO32MB)
+#if defined(CONFIG_PSIONW_5MXPRO16MB) || defined(CONFIG_PSIONW_5MXPRO24MB) || defined(CONFIG_PSIONW_5MXPRO32MB)
 	/* Set the memory wait states to PCMCIA */
 	// FIXME5MX: Should be checked...
 	psionw_writel(0x93930002, MEMCFG1);
diff -ur linux24-psionw/include/asm-arm/hardware/psionw.h linux24-psionw-16pro/include/asm-arm/hardware/psionw.h
--- linux24-psionw/include/asm-arm/hardware/psionw.h	2000-01-01 00:00:00.000000000 +0100
+++ linux24-psionw-16pro/include/asm-arm/hardware/psionw.h	2006-07-29 22:46:04.000000000 +0200
@@ -111,7 +111,7 @@
 /*
  * ETNA PCMCIA Controller
  */
-#if defined(CONFIG_PSIONW_5MXPRO24MB) || defined(CONFIG_PSIONW_5MXPRO32MB)
+#if defined(CONFIG_PSIONW_5MXPRO16MB) || defined(CONFIG_PSIONW_5MXPRO24MB) || defined(CONFIG_PSIONW_5MXPRO32MB)
 #define ETNA_P_BASE	0x50000000	/* ETNA controller physical base */
 #else
 #define ETNA_P_BASE	0x20000000	/* ETNA controller physical base */
@@ -150,7 +150,7 @@
 #define ETNA_CLEAR_MASK	0xff		/* Clear ETNA interrupt */
 #define	ETNA_CF_IRQ	0x01	        /* Enable ETNA Interrupt */
 
-#if defined(CONFIG_PSIONW_5MXPRO24MB) || defined(CONFIG_PSIONW_5MXPRO32MB)
+#if defined(CONFIG_PSIONW_5MXPRO16MB) || defined(CONFIG_PSIONW_5MXPRO24MB) || defined(CONFIG_PSIONW_5MXPRO32MB)
 #define CF1_P_BASE	0x30000000	/* PCMCIA physical base */
 #else
 #define CF1_P_BASE	0x40000000	/* PCMCIA physical base */
